src/reducers/dataLoggerReducer.js

```diff
@@ -4,21 +4,55 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
  */
 
+import { unit } from 'mathjs';
+
 import {
     getDuration,
+    getMaxBufferSize,
     getSampleFreq,
     setDuration as persistDuration,
+    setMaxBufferSize as persistMaxBufferSize,
     setSampleFreq as persistSampleFreq,
 } from '../utils/persistentStore';
 
-const ranges = [
-    { name: 'days', multiplier: 24 * 60 * 60, min: 7, max: 500 }, // 1Hz
-    { name: 'days', multiplier: 24 * 60 * 60, min: 1, max: 50 }, // 10Hz
-    { name: 'hours', multiplier: 60 * 60, min: 6, max: 120 }, // 100Hz
-    { name: 'hours', multiplier: 60 * 60, min: 1, max: 12 }, // 1kHz
-    { name: 'minutes', multiplier: 60, min: 10, max: 72 }, // 7.7-10kHz
-    { name: 'seconds', multiplier: 1, min: 60, max: 432 }, // 100kHz
-];
+/**
+ * [Get ranges with max field adjusted for the current maxBufferSize]
+ * @param {Number} maxBufferSize []
+ * @param {Object} ranges []
+ * @returns {Object} [with max field adjusted for max sampling buffer size]
+ */
+const getAdjustedRanges = (maxBufferSize, ranges) => {
+    // Maximum number of elements that the current maxBufferSize could initialize.
+    const bytesPerElement = 4;
+    const maxNumberOfElements = Math.floor(
+        unit(maxBufferSize, 'MB').to('byte').toNumber() / bytesPerElement
+    );
+
+    ranges.forEach((_range, index) => {
+        // Find out how many seconds, minutes, hours or days is required to fill the array with any given frequency.
+        const { multiplier, frequency } = _range;
+        ranges[index].max = Math.floor(
+            maxNumberOfElements / (multiplier * frequency)
+        );
+    });
+    return ranges;
+};
+
+const initialMaxBufferSize = getMaxBufferSize(173);
```

Comment:
Should we describe this apparently arbitrary number `173`?

Reply:
How would you purpose we describe it? Maybe we can increase it to 200MB, or is that too much?

Reply:
Sounds good to me.